Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +7,217 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            
+
+            var firstAccount = peopleAndBalances[0].Split(", ");
+            var firstBalance = Lesson7.StringArrayToFloatArray(firstAccount[1..]);
+
+            var names = new string[] {firstAccount[0]};
+            var highestBalance = Lesson7.MaxFloat(firstBalance);
+
+            for (int i = 1; i < peopleAndBalances.Length; i++)
+            {
+                var currentAccount = peopleAndBalances[i].Split(", ");
+                var currentBalance = Lesson7.StringArrayToFloatArray(currentAccount[1..]);
+
+                var currentName = currentAccount[0];
+                var currentHighest = Lesson7.MaxFloat(currentBalance);
+
+                // if same as others add it to the list
+                if (currentHighest == highestBalance)
+                {
+                    names = Lesson7.AppendString(names, currentName);
+                }
+
+                if (currentHighest > highestBalance)
+                {
+                    names = new string[] {currentName};
+                    highestBalance = currentHighest;
+                }
+            }
+
+            string returnNames;
+            if (names.Length == 1)
+            {
+                returnNames = names[0];
+            }
+            else
+            {
+                returnNames = string.Join(", ", names[..^1]);
+                returnNames += $" and {names[^1]}";
+            }
+
+            return $"{returnNames} had the most money ever. ¤{highestBalance}.";
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+            string[] firstAccount;
+            float[] firstBalance;
+            int firstIndex = 0;
+            do
+            {
+                firstAccount = peopleAndBalances[firstIndex].Split(", ");
+                firstBalance = Lesson7.StringArrayToFloatArray(firstAccount[1..]);
+                firstIndex++;
+            } while (firstBalance.Length < 2 && firstIndex < peopleAndBalances.Length);
+
+
+            if (firstIndex == peopleAndBalances.Length && firstBalance.Length < 2)
+            {
+                return "N/A.";
+            }
+
+
+            string[] names = new string[] { firstAccount[0] };
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5001581)